lib/plugins/aws/package/compile/functions/index.test.js

```diff
@@ -2831,4 +2810,207 @@ describe('AwsCompileFunctions #2', () => {
+      // therefore not be included in the version id digest
+      const firstVersionId = firstCfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref;
+      const firstS3Key = firstCfTemplate.Resources.TestLayerLambdaLayer.Properties.Content.S3Key;
+
+      const data = await runServerless({
+        cwd: servicePath,
+        cliArgs: ['package'],
+        awsRequestStubMap: {
+          CloudFormation: {
+            describeStacks: { Stacks: [{ Outputs: [{ OutputKey: 'test' }] }] },
+          },
+        },
+      });
+
+      expect(firstS3Key).to.not.equal(
+        data.cfTemplate.Resources.TestLayerLambdaLayer.Properties.Content.S3Key
+      );
+      expect(firstVersionId).to.equal(
+        data.cfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref
+      );
+    });
+
+    it('should ignore properties order when generating layer version id', async () => {
+      const firstVersionId = firstCfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref;
+
+      await updateConfig({
+        functions: {
+          foo: { layers: [{ Ref: 'TestLayerLambdaLayer' }], handler: 'index.handler' },
+        },
+      });
+
+      const data = await runServerless({
+        cwd: servicePath,
+        cliArgs: ['package'],
+        awsRequestStubMap: {
+          CloudFormation: {
+            describeStacks: { Stacks: [{ Outputs: [{ OutputKey: 'test' }] }] },
+          },
+        },
+      });
+
+      expect(firstVersionId).to.equal(
+        data.cfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref
+      );
+    });
+
+    it('should create different lambda version id for different property keys (but no different values)', async () => {
+      const firstVersionId = firstCfTemplate.Outputs.FooBarLambdaFunctionQualifiedArn.Value.Ref;
+
+      await updateConfig({
+        functions: {
+          fooBar: { handler: 'index.handler', provisionedConcurrency: 1 },
+        },
+      });
+
+      const data = await runServerless({
+        cwd: servicePath,
+        cliArgs: ['package'],
+        awsRequestStubMap: {
+          CloudFormation: {
+            describeStacks: { Stacks: [{ Outputs: [{ OutputKey: 'test' }] }] },
+          },
+        },
+      });
+
+      expect(firstVersionId).to.equal(
+        data.cfTemplate.Outputs.FooBarLambdaFunctionQualifiedArn.Value.Ref
+      );
+    });
+
+    it('should create same version id for same layer zip content', async () => {
+      const firstVersionId = firstCfTemplate.Outputs.FooLambdaFunctionQualifiedArn.Value.Ref;
+      const originalLayer = path.join(servicePath, 'testLayer');
+      const sourceChangeLayer = path.join(servicePath, 'extra_layers', 'testLayerDuplicate');
+      const backupLayer = path.join(servicePath, 'extra_layers', 'testLayerBackup');
+
+      await fse.rename(originalLayer, backupLayer);
+      await fse.rename(sourceChangeLayer, originalLayer);
```

Comment:
To test it properly I believe we should change the layer path in a config (so point a file at different location with a same content).

Here we're replacing file with file of very same content, which technically brings no change in environement. which seems equal to not do any renames.

Reply:
Yes re-reading this I think it is redundant. Changing the path in the config would also cause a version id change as the config is part of the hash. I think what I was trying to do here is test that if nothing is changed in the layer config (including source) then the version id stays the same.

This is kind of already covered in the `should ignore changing character of S3Key paths when generating layer version id`, but I've kept the test and simplified it to be `should create same version id when layer source and config are unchanged` and just ran `runServerless` twice. The benefit of keeping it would be to make clear cases where the S3Key test failed due to a more general and S3Key-unrelated issue.